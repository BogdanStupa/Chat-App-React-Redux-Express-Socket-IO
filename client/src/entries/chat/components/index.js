import UserInfoComponent from "./user-info-component";
import InputSearchComponent from "./input-search";

export {
    UserInfoComponent,
    InputSearchComponent
};