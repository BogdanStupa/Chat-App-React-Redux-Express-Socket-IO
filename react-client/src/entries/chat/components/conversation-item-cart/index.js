import React from "react";
import { useSelector } from "react-redux";
import { 
    conversationItemSelector,
    currentConversationItemSelector,
    unreadMessagesSelector
 } from "selectors";
import ConversationItemCart from "./conversation-item-cart-component";


const ConversationItemCartContainer = props => {
    const {
        onClickItem,
        id
    } = props;
    const activeConversationId = useSelector(state => currentConversationItemSelector(state,"conversationId"));
    const item = useSelector(state => conversationItemSelector(state, id)) || {};
    const unreadMessages = useSelector(state => unreadMessagesSelector(state,id));
    
    return (    
        <ConversationItemCart
            id={id}
            unreadMessages={unreadMessages}
            profile={{
                label: item.partner.nickname,
                backgroundColor: item.partner.profileColor,
                width: 45,
                height: 45,
                fontSize: 24
            }}
            message={{
                messageText: item.partner.lastMessage ? item.partner.lastMessage.message : "no messages yet...",
                dateTime: item.partner.lastMessage ? item.partner.lastMessage.dateTime : null
            }} 
            partnerId={item.partner.partnerId}
            unreadMessages={item.unreadMessages}
            nickname={item.partner.nickname}
            onClickItem={onClickItem}
            isActive={item._id === activeConversationId ? true : false}
        /> 
    );
}   

export default ConversationItemCartContainer;