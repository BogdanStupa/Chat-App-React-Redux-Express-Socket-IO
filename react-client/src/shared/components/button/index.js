import React, { forwardRef } from "react";
import { compose } from "redux";
import classNames from "classnames";


const ButtonComponent = forwardRef((props, ref) => {
    const {
        text,
        primary,
        type,
        link,
        isFetching,
        marginTop,
        height,
        width,
        textDecoration,
        onClick,
        children
    } = props;

    const styleButton = classNames({
        btn: true,
        "btn--default": true && !link && !isFetching,
        "btn--link": link,
        "btn--disabled": isFetching,
        "btn--primary": !isFetching && primary
    });

    return (
        <button
            className={styleButton}
            type={type}
            disabled={isFetching}
            onClick={onClick}
            style={{
                marginTop,
                height,
                width,
                textDecoration
            }}
            ref={ref}
        >
            {
                isFetching 
                ? "loading"
                : text
            }
            {
                children
            }
        </button>
        
    );
});

export default compose(
    React.memo
)(ButtonComponent);;
