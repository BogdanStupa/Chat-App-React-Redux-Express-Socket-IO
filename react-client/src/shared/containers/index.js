import AppContainer from "./app";
import AuthContainer from "./auth";
import PrivateRoute from "./private";
export {
    AppContainer,
    AuthContainer,
    PrivateRoute
};