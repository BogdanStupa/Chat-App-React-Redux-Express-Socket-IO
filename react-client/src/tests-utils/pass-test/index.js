import React, { Component } from "react";
import { shalow } from "enzyme";

export default (Component) => {
    describe("Base test",() => {
        it("should pass through props to the component", () => {
            const customAttributeName = `data-test${Math.random()}`;
            const props = {
                [customAttributeName]: "foo"
            }
            const wrapper = shalow(<Component {...props}/>);
            expect(wrapper.first().props()[customAttributeName]).toEqual("foo");
        });
    });
} 