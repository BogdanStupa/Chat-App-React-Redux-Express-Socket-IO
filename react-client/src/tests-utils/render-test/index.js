import React from "react";
import { shallow } from "enzyme";


export default (name,Component) => {
    describe(`${name}it should be renderd`,() => { 
        const wrapper = shallow(Component);
        it("is defined", () => {
            expect(Component).toBeDefined();
        });
        it("renders the component", () => {
            expect(false).toEqual(wrapper.isEmptyRender());
        });
        it("matches the snapshots",() =>{
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
    
}
