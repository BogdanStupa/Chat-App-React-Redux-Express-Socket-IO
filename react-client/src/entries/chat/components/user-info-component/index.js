import React from "react";
import classNames from "classnames";
import { ProfileCircle } from "shared/components";

const UserInfoComponent = props => {
    const {
        profile,
        title,
        rowMode
    } = props;
    
    const containerStyles = classNames({
        "user-info": true,
        "column-mode": !rowMode,
        "row-mode": rowMode
    });

    return (
        <div className={containerStyles}>
            <ProfileCircle
                profile={profile}
            />
            <p 
                className="user-info-nickname"
                style={{
                    fontSize: title.fontSize,
                    color: title.color,
                    fontWeight:"bold"
                }}
            >
                    {title.nickname}
            </p>
        </div>
    );
};

export default UserInfoComponent;