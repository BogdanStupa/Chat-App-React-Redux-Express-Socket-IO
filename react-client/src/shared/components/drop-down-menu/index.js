import React, { useState, useRef } from "react";
import { usePopper } from "react-popper";
import classNames from "classnames";
import onClickOutside from "react-onclickoutside";
import { ButtonComponent } from "shared/components";



function DropDownMenuComponent(props){
    const {
        icon,
        options 
    } = props;

    const [isOpen, setIsOpen] = useState(false);
    const boxRef = useRef();
    const tooltipRef = useRef();
    
    const { styles, attributes } = usePopper(boxRef.current, tooltipRef.current, {
        placement:"bottom-end"
    });

    const onClickHandler = () => setIsOpen(!isOpen);

    DropDownMenuComponent.handleClickOutside = () => setIsOpen(false);

    return (    
        <div className="drop-down-wrapper">
            <div ref={boxRef}>
                <ButtonComponent
                    link
                    onClick={onClickHandler}
                >
                    { icon }
                </ButtonComponent>
            </div>
            <div 
                className={classNames("tooltip",{ "tooltip-hidden": !isOpen })}
                ref={tooltipRef} 
                style={styles.popper} 
                { ...attributes.popper }
            >
                {
                    options.map((item, index) => (
                        <li key={index} >
                            <ButtonComponent
                                link
                                textDecoration="none"
                                onClick={item.onClick}
                            >
                                { item.text }
                            </ButtonComponent>
                        </li>
                        
                    ))
                }   
            </div>        
        </div>
    );
}


const clickOutsideConfig = {
    handleClickOutside: () => DropDownMenuComponent.handleClickOutside
  };

export default onClickOutside(DropDownMenuComponent, clickOutsideConfig);