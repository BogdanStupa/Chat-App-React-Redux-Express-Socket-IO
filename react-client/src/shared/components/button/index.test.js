import React from "react";
import { ButtonComponent } from "shared/components";
import { shallow } from "enzyme";
import { componentCheckOnRender } from "tests-utils";
import { getByText, render } from '@testing-library/react';



describe("<ButtonComponent />:", () => { 
    componentCheckOnRender("ButtonComponent",<ButtonComponent/>);
    
    describe("should working correctly:", () => {
        const onButtonClick = jest.fn();
        const modelI = {
            text: "testeButton",
            type: "submit",
            isFetching: false,
            onClick: onButtonClick
        };
        const modelII = {
            text: "testeButton",
            type: "submit",
            isFetching: true,
            onClick: onButtonClick
        };
        const getWrapper = (model) => {
            return shallow(<ButtonComponent { ...model } />)  
        };

        it("has a button component", () => {
            const wrapper = getWrapper(modelI);
            expect(wrapper.find("button").length).toBe(1);
        });

        it("has (loading) whith isFetching === true, show loading",() => {
            const wrapper = getWrapper(modelII);
            expect(wrapper.text()).toEqual("loading");
        });

        it("has no (loading) with isFetching === false, show text",() => {
            const wrapper = getWrapper(modelI);
            expect(wrapper.text()).toEqual(modelI.text);
        });

        it("onClick button is working",() => {
            const wrapper = getWrapper(modelI);
            wrapper.simulate("click");
            expect(onButtonClick).toHaveBeenCalledTimes(1);
        });
    
    });
});