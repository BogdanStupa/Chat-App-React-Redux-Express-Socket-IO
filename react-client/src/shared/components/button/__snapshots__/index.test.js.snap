// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonComponent />: ButtonComponentit should be renderd matches the snapshots 1`] = `"<button class=\\"btn btn--default\\"></button>"`;
